tests/test_install_mender_sh.py

```diff
@@ -0,0 +1,153 @@
+        """ We need to access the toplevel host's port to curl the script.
+            'localhost' of course doesn't work, but qemu runs on docker in 'net=host' mode, 
+            so the host's address is simply the default route.
+        """
+        result = setup_tester_ssh_connection.run(
+            "ip route | grep default | awk '{print $3}'"
+        )
+        return result.stdout.strip()
+
+    @pytest.mark.usefixtures("setup_test_container_f")
+    @pytest.mark.parametrize("channel", ["", "stable", "experimental"])
+    @pytest.mark.skip(
+        reason="no mender-shell in 'stable'; in 'experimental' depends on client 2.5.0 which is not there yet"
+    )
+    def test_default(
+        self,
+        script_server,
+        setup_tester_ssh_connection_f,
+        mender_dist_packages_versions,
+        mender_version,
+        channel,
+    ):
+        """ Default, no arg install installs mender-client and mender-shell (stable).
+        """
+        localhost = self._get_localhost_ip(setup_tester_ssh_connection_f)
+
+        if channel != "":
+            channel = "-c " + channel
+
+        setup_tester_ssh_connection_f.run(
+            "curl http://{}:{}/install-mender.sh | sudo bash -s -- {}".format(
+                localhost, SSERVER_PORT, channel
+            )
+        )
+
+        setup_tester_ssh_connection_f.run("dpkg -s mender-client")
+        setup_tester_ssh_connection_f.run("dpkg -s mender-shell")
+
+    @pytest.mark.usefixtures("setup_test_container_f")
+    @pytest.mark.parametrize("channel", ["", "stable", "experimental"])
+    def test_client(
+        self,
+        script_server,
+        setup_tester_ssh_connection_f,
+        mender_dist_packages_versions,
+        mender_version,
+        channel,
+    ):
+        localhost = self._get_localhost_ip(setup_tester_ssh_connection_f)
+
+        if channel != "":
+            channel = "-c " + channel
+
+        setup_tester_ssh_connection_f.run(
+            "curl http://{}:{}/install-mender.sh | sudo bash -s -- {} mender-client".format(
+                localhost, SSERVER_PORT, channel
+            )
+        )
+
+        setup_tester_ssh_connection_f.run("dpkg -s mender-client")
+
+        # piggyback misc cmdline tests to save an extra container run
+        # help
+        setup_tester_ssh_connection_f.run(
+            "curl http://{}:{}/install-mender.sh | bash -s -- -h".format(
+                localhost, SSERVER_PORT
+            )
+        )
+        assert "usage:" in res.stdout
+
+        # invalid arg/module
+        try:
+            setup_tester_ssh_connection_f.run(
+                "curl http://{}:{}/install-mender.sh | bash -s -- unknown".format(
+                    localhost, SSERVER_PORT
+                )
+            )
+        except Exception as ex:
+            assert "Unsupported argument: `unknown`" in str(ex)
+        else:
+            assert False, "command should fail"
```

Comment:
This works, but a cleaner way is using the `warn` parameter of Fabric (obs, this is not available in integration tests but only on the other frameworks)

Something in the lines of:
```suggestion
        setup_tester_ssh_connection_f.run(
            "curl http://{}:{}/install-mender.sh | bash -s -- unknown".format(
                localhost, SSERVER_PORT
            ),
            warn=True
        )
        assert result.exited == 1
        assert "Unsupported argument: `unknown`" in result.stderr
```